File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        
        int[] a = new int[nums.Length];
        int[] b = new int[nums.Length];
        int[] ans = new int[nums.Length];
        
        a[0] = nums[0];
        b[nums.Length-1] = nums[nums.Length-1];

        for (int i = 1; i < nums.Length; i++)
        {
            a[i] = nums[i] *  a[i - 1];
            b[nums.Length-1-i] = nums[nums.Length-i-1] *  b[nums.Length-i];
        }

        ans[0] = b[1];
        ans[ans.Length-1] = a[ans.Length-2];

        for (int i = 1; i < ans.Length-1; i++)
        {
            ans[i] = a[i-1] *  b[i+1];
        }
        
        return ans;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-3.java
class Solution {
    public static List<List<Integer>> threeSum(int[] nums) {

        Arrays.sort(nums);
        ArrayList<List<Integer>> res = new ArrayList<>();

        for (int i = 0; i < nums.length-1; i++) {
            // if (nums[i] == nums[i+1]) continue;

            if (i > 0 && nums[i] == nums[i - 1]) {
                continue;
            }

            int l = i+1;
            int r = nums.length-1;

            while(l<r){
                if (nums[i] + nums[l] + nums[r] == 0) {
                    res.add(List.of(nums[i],nums[l],nums[r]));
                    l++;
                    r--;   

                        while (l < r && nums[l] == nums[l-1]) {
                            l++;
                        }

                        while (l < r && nums[r] == nums[r+1]) {
                            r--;
                        }
                }
                
                else if (nums[i] + nums[l] + nums[r] < 0 ) {
                    l++;
                }
                else if (nums[i] + nums[l] + nums[r] > 0 ) {
                    r--;
                }
            }
        }
        return res;
    }
}
